src/main/scala/com/databricks/spark/avro/SchemaConverters.scala

```diff
@@ -112,6 +119,80 @@ private object SchemaConverters {
   }
 
   /**
+   * This function constructs a converter function that will be used to convert avro types to their
+   * corresponding sparkSQL representations.
+   */
+  private[avro] def createConverterToSQL(schema: Schema): Any => Any = {
+    schema.getType match {
+      // Avro strings are in Utf8, so we have to call toString on them
+      case STRING | ENUM => (item: Any) => if (item == null) null else item.toString
+      case INT | BOOLEAN | DOUBLE | FLOAT | LONG => identity
+      // Byte arrays are reused by avro, so we have to make a copy of them.
+      case FIXED => (item: Any) => item match {
+        case null => null
+        case fixed: Fixed => fixed.bytes().clone()
+      }
+      case BYTES => (item: Any) => item match {
+        case null => null
+        case bytes: ByteBuffer =>
+          val javaBytes = new Array[Byte](bytes.remaining)
+          bytes.get(javaBytes)
+          javaBytes
+      }
+      case RECORD =>
+        val fieldConverters = schema.getFields.map(f => createConverterToSQL(f.schema))
+        (item: Any) => item match {
+          case null => null
+          case record: GenericRecord =>
+            Row.fromSeq(fieldConverters.zipWithIndex.map { case (conv, index) =>
```

Comment:
I'd consider avoiding functional programming here since this is in the critical path.